src/prop.ts

```diff
@@ -151,7 +151,7 @@ const baseProp = (rawOptions, Type, target, key, isArray = false) => {
 };
 
 export const prop = (options: PropOptionsWithValidate = {}) => (target: any, key: string) => {
-  const Type = Reflect.getMetadata('design:type', target, key);
+  const Type = (Reflect as any).getMetadata('design:type', target, key);
```

Comment:
Yes, it's weird and I don't know why compiler complain about it. However it should be an issue since #47 will make `reflect-metadata` package out of the deps, so casting to any in this place will be necessary.